packages/drivers/routerlicious-urlResolver/src/urlResolver.ts

```diff
@@ -13,7 +13,8 @@ import {
     IUrlResolver,
 } from "@fluidframework/driver-definitions";
 import { IUser, ScopeType } from "@fluidframework/protocol-definitions";
-import { generateToken, IAlfredTenant } from "@fluidframework/server-services-client";
+import { IAlfredTenant } from "@fluidframework/server-services-client";
+import { generateToken } from "@fluidframework/server-services-utils";
```

Comment:
This causes a layer violation. @tanviraumi - is it safe to use jsrsasign here or do we do something else? generateToken is no longer in server-services-client

Reply:
I see there is only 1 usage within the repo - https://github.com/microsoft/FluidFramework/blob/f8cef2c86044f9d2244a9dfe47052e383ca8f525/server/gateway/src/routes/api/api.ts

@jatgarg - I noticed you added this API. We need the client packages to not depend on jsonwebtoken. Is it safe for us to just use generateTokenDirectly from gateway/api.ts rather than depending on a client package for that?

Reply:
I would wait for @jatgarg to confirm my thoughts but this is not the right place to sign a token. Token signing should be done outside the driver. So my recommendation is to make the second parameter in RouterliciousUrlResolver required by removing the "| undefined". Then you can remove the getR11sToken call and also the function. Gateway builds separately so next time when gateway gets updated, people will use the new function in services-utils package. It would be nice if you can put a comment in gateway in this PR.

@jatgarg Thoughts?

Reply:
@jatgarg  - can you take a look, I plan to merge this today

Reply:
Yes @tanviraumi idea looks good to me to make the argument compulsory and not generate token inside the resolver.